zpay32/encode.go

```diff
@@ -164,6 +164,17 @@ func writeTaggedFields(bufferBase32 *bytes.Buffer, invoice *Invoice) error {
 		}
 	}
 
+	if invoice.Metadata != nil {
+		base32, err := bech32.ConvertBits(invoice.Metadata, 8, 5, true)
```

Comment:
`writeBytes32` can be used here instead.

Reply:
`writeBytes32` always writes 32 bytes, where as the invoice metadata is variable length?